library/core/src/ptr/mod.rs

```diff
@@ -998,14 +998,15 @@ pub unsafe fn write<T>(dst: *mut T, src: T) {
 /// ```
 #[inline]
 #[stable(feature = "ptr_unaligned", since = "1.17.0")]
-pub unsafe fn write_unaligned<T>(dst: *mut T, src: T) {
+#[rustc_const_unstable(feature = "const_ptr_write", issue = "none")]
+pub const unsafe fn write_unaligned<T>(dst: *mut T, src: T) {
     // SAFETY: the caller must guarantee that `dst` is valid for writes.
     // `dst` cannot overlap `src` because the caller has mutable access
     // to `dst` while `src` is owned by this function.
     unsafe {
         copy_nonoverlapping(&src as *const T as *const u8, dst as *mut u8, mem::size_of::<T>());
     }
-    mem::forget(src);
+    intrinsics::forget(src);
```

Comment:
I guess you changed this for consistency? Fine for me, but please then also add the comment here.

Reply:
Oh, sorry did not realize that. Sure thing :)

Would you prefer if I `git commit --amend` those (hopefully) last few changes

Reply:
Sure, just amend this.

Reply:
Done. I also moved `intrinsics::forget(src);` inside the `unsafe` block